### DISCUSSION TITLE
Blazor Server to Client - Client to Server Variable Read and Write

### DISCUSSION THREAD

**Discussion by Beblia** [dotnet/aspnetcore#23450] [General]
I have a blazor web assembly project with blazor server running together. Is there a way to save data on the client side and then be able to read it from the server? I tried using query but it never updates on the server. I have also tried Sessions, LocalStorage but none can be read from the server side. The data im storing does not need authentication or any security at all. Best case is if I can just send the data in a query string but blazor server does not read the query string that I set from blazor client. If you can please help me out. Maybe post some psedo code that would be perfect. Thanks.

**Comment by Beblia** [comment]
Please help anyone. My project is dead right now without this fix. Thank you.

**Comment by JRC86** [comment]
Do you have a repo ?

**Reply by Beblia** [reply]
Not really. Basically I have 2 projects in my solution. One acts as the server for Blazor and the other as client web assembly. I want to save data in client and have that data be readable in the server side. Also I want to save data on the server and read it on the client side. Right now I cant do any of that. I tried global variables, I tryed cookies. Nothing works. If i save data on the client then only the client sees the saved data such as a cookie but I cant read the new cookie data from the server. Seems that client and server are running in their own sandbox and no data is shared between them. I need to share data otherwise this project cant go any further.
